### ISSUE TITLE
The ldp:inbox and ietf:cite-as are intepreted as a literals and not as resources by JSON-LD processors

### ISSUE THREAD

**Issue by phochste** [antleaf/notify-implementation#6]
When the COAR notification are processed by JSON-LD processors (such as Jena or the JSON-LD playground), then the `ldp:inbox` and `ietf:cite-as` are interpeted as RDF literals and not as resources (IRI).

**Steps to replicate**

1. Open a Notify example in the JSON-LD playground:

https://json-ld.org/playground/#startTab=tab-expanded&json-ld=https%3A%2F%2Fgist.githubusercontent.com%2Fphochste%2F549acb93af22ff91e4ab821443de8517%2Fraw%2F0e388ca7b82259f089919f1986b9dea6d5d077ec%2Foffer.jsonld

2. Click on the **N-Quads** tab
3. Check the ldp:inbox and ietf:cite-as resources

```
<https://origin-system.org/resources/0021> <http://www.iana.org/assignments/relation/cite-as> "https://doi.org/10.4598/12123487" 
<https://origin-system.org> <http://www.w3.org/ns/ldp#inbox> "https://origin-system.org/inbox/" .
<https://target-system.org> <http://www.w3.org/ns/ldp#inbox> "https://target-system.org/inbox/" .
```

**Expected result**

The `ietf:cite-as` and `ldp:inbox` resources should be IRI-s:

```
<https://origin-system.org/resources/0021> <http://www.iana.org/assignments/relation/cite-as> <https://doi.org/10.4598/12123487> .
<https://origin-system.org> <http://www.w3.org/ns/ldp#inbox> <https://origin-system.org/inbox/> .
<https://target-system.org> <http://www.w3.org/ns/ldp#inbox> <https://target-system.org/inbox/> .
```

**Comment by phochste** [comment]
```
  "@context": [
    "https://www.w3.org/ns/activitystreams",
    {
      "ietf:cite-as": {
        "@id" : "http://www.iana.org/assignments/relation/cite-as",
        "@type" : "@id"
      } ,
      "ldp:inbox": {
        "@id": "http://www.w3.org/ns/ldp#inbox" ,
        "@type": "@id"
      }
    }
  ] ...
```

This provides the desired effect.

**Comment by paulwalk** [comment]
Are there examples of this in other contexts?

**Comment by phochste** [comment]
I need to check in other contexts. I know the IIIF uses similar tricks. Maybe Robert Sanderson is a the best place to ask. 

What I found out is that both [https://www.w3.org/ns/activitystreams](https://www.w3.org/ns/activitystreams.jsonld) and [http://www.w3.org/ns/ldp](http://www.w3.org/ns/ldp.jsonld) define the exactly the same `http://www.w3.org/ns/ldp#inbox` property. 

We are not required to add `"ldp": "http://www.w3.org/ns/ldp#"` in the `@context` and every `ldp:inbox` can could written as `inbox` (using the default AS2 namespace) and we get exactly the required effect.

This context would do the trick: 

```(JSON)
{
  "@context": [
    "https://www.w3.org/ns/activitystreams" ,
    {
      "ietf" : "http://www.iana.org/assignments/relation"
    }
  ] ,
 "id": "urn:uuid:0370c0fb-bb78-4a9b-87f5-bed307a509dd",
 "type": [ "Offer"  ],
 "origin": {
    "id": "https://origin-system.org",
    "inbox": "https://origin-system.org/inbox/",
    "type": [ "Service" ]
 }
}
```

which is equivalent to:

```
<https://origin-system.org> <http://www.w3.org/1999/02/22-rdf-syntax-ns#type> <https://www.w3.org/ns/activitystreams#Service> .
<https://origin-system.org> <http://www.w3.org/ns/ldp#inbox> <https://origin-system.org/inbox/> .
<urn:uuid:0370c0fb-bb78-4a9b-87f5-bed307a509dd> <http://www.w3.org/1999/02/22-rdf-syntax-ns#type> <https://www.w3.org/ns/activitystreams#Offer> .
<urn:uuid:0370c0fb-bb78-4a9b-87f5-bed307a509dd> <https://www.w3.org/ns/activitystreams#origin> <https://origin-system.org> .
```

There is still an issue with the `ietf` namespace because it doesn't define a JSON-LD schema. JSON-LD processors can't
know that the `ietf:cite-as` (which is used in the examples) is an IRI resource and will treat it as a string Literal. One way is to make this explicit in the context or create a COAR JSON-lD schema document ourselves, which adds the properties not available in AS2:

```
{
  "@context": [
    "https://www.w3.org/ns/activitystreams" ,
    {
      "ietf" : "http://www.iana.org/assignments/relation/" ,
      "ietf:cite-as": {
        "@id" : "ietf:cite-as",
        "@type" : "@id"
      } 
    }
  ] 
```

**Comment by paulwalk** [comment]
> or create a COAR JSON-lD schema document ourselves ( adding the properties not available in AS2)

If I've understood your suggestion correctly, we would follow the same patter:

```json
"ietf:cite-as": {
        "@id" : "ietf:cite-as",
        "@type" : "@id"
      } 
```

and this would give the advantage of defining this once in the schema file (which we then include in the context), rather than in every payload.

Is that right?

**Comment by phochste** [comment]
Yes , e.g. publish a document `https://notify.coar-repositories.org/ns/activity`

Which returns a JSON-LD schema like :

```(JSON)
{
  "@context" : {
    "nat": "http://purl.org/coar/notify_activity_type/" ,
    "nrr": "http://purl.org/coar/notify_resource_relations/" ,
    "ietf": "http://www.iana.org/assignments/relation/" ,
    "ietf:cite-as": {
        "@type" : "@id"
    }
  }
}
```

With that all COAR notifications can look like: 

```(JSON)
{
  "@context": [
    "https://www.w3.org/ns/activitystreams",
    "https://notify.coar-repositories.org/ns/activity"
  ],
  "id" : " ...." ,
  "type" : [ ... ] ,
  "actor" : ...
```

I think we need to discuss this a bit further what could be add to this COAR JSON-LD schema. I see in the examples:

- nat:review-request
- nat:review-success
- nat:endorsement-success
- nrr:reviews
- nrr:endorses

The creation of a COAR JSON-LD schema is helpful, but should be discussed in detail. There is also a naming convention (e.g. classes have typically camelcase names *ReviewRequest* and properties have lowercase names *reviews*).

**Comment by paulwalk** [comment]
Right, understood - thanks. I had hoped that we could proceed with only needing to create some simple vocabularies (using something like SKOS), but it seems that we probably do need to establish a COAR Notify schema.

I think we should establish a minimal schema quickly - we don't have the luxury of much time right now. A minimal schema can always be added to (much easier than changing/deleting properties).

**Comment by phochste** [comment]
Ok I'll suggest a schema via a plull request

**Comment by hvdsomp** [comment]
Have a look at the [JSON-LD context](https://datatracker.ietf.org/doc/html/draft-ietf-httpapi-linkset-00#appendix-B) that was provided by Stian as part of the Link Set Internet Draft effort. It also contains a declaration for the IANA link relation type registry terms. I wonder whether this one is correct ...
